brave/brave-ios
# Fix #5815: Wallet networks fail to load, causing empty network picker

## Summary of Changes
- Restructure `AccountActivityStore` to set itself up in a `setup()` function called when the view appears instead of at initialization of the store.
- The main issue is the wrapper for `JsonRpcService` holds onto a `mojo::Remote<brave_wallet::mojom::JsonRpcService>` (that is bound to the `JsonRpcService` implementation) was being disconnected. The cause is currently unknown, however from debugging this seems to work around the issue (no more mojom disconnects for `JsonRpcService`) by delaying the calls to the service until the view appears.

This pull request fixes #5815

## Submitter Checklist:


## Test Plan:
To reproduce, on any branch but `wallet-networks-fail-load`:
1. Open wallet and switch to Solana network
2. Open `app.1inch.io` dapp and connect your wallet
3. Tap the wallet icon to open/close the panel. The bug occurs occasionally, you will see an empty network picker.
4. Once you get the empty network picker, tap it to show the network picker modal with no networks inside.

To test the fix, follow same steps or steps from the issue and try to reproduce.


## Screenshots:
Bug:

https://user-images.githubusercontent.com/5314553/184167483-e2f90aec-efee-4fc4-a158-b4c02a73b4e7.mp4

Fixed:

https://user-images.githubusercontent.com/5314553/184167525-e664c686-2c10-4f8a-8731-eaf3f92da5e1.mp4


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
